src/platform/intel/cavs/lib/pm_memory.c

```diff
@@ -0,0 +1,209 @@
+		/* Set the bits for all powered banks operations
+		 * depend whether we turn on or off the memory
+		 */
+		operation_data->ebb_new_mask0 =
+			operation_data->ebb_current_mask0;
+		operation_data->ebb_new_mask1 = bit ?
+			operation_data->ebb_current_mask1
+			| operation_data->change_mask1 :
+			operation_data->ebb_current_mask1
+			& ~operation_data->change_mask1;
+
+	/* second check if managing spans above and below EBB_SEGMENT_SIZE
+	 * meaning we have to handle both operation_data for both segments
+	 */
+	} else if (operation_data->start_bank_id > EBB_SEGMENT_SIZE_ZERO_BASE
+		   && operation_data->ending_bank_id
+		   < EBB_SEGMENT_SIZE_ZERO_BASE) {
+		operation_data->ebb_new_mask0 = bit ?
+			operation_data->ebb_current_mask0
+			| operation_data->change_mask0 :
+			operation_data->ebb_current_mask0
+			& ~operation_data->change_mask0;
+		operation_data->ebb_new_mask1 = bit ?
+			operation_data->ebb_current_mask1
+			| operation_data->change_mask1 :
+			operation_data->ebb_current_mask1
+			& ~operation_data->change_mask1;
+
+	/* if only first segment needs changes */
+	} else {
+		operation_data->ebb_new_mask0 = bit ?
+			operation_data->ebb_current_mask0
+			| operation_data->change_mask0 :
+			operation_data->ebb_current_mask0
+			& ~operation_data->change_mask0;
+		operation_data->ebb_new_mask1 =
+			operation_data->ebb_current_mask1;
+	}
+}
+
+/**
+ * \brief Write calculated masks to HW registers
+ * \param[in] ebb_data All masks data with masks to write.
+ */
+static void write_new_masks_and_check_status(struct ebb_data *operation_data)
+{
+	uint32_t status, delay_count = 256;
+	/* HSPGCTL, HSRMCTL use reverse logic - 0 means EBB is power gated */
+	io_reg_write(HSPGCTL0, ~operation_data->ebb_new_mask0);
+	io_reg_write(HSRMCTL0, ~operation_data->ebb_new_mask0);
+	io_reg_write(HSPGCTL1, ~operation_data->ebb_new_mask1);
+	io_reg_write(HSRMCTL1, ~operation_data->ebb_new_mask1);
+
+	/* Query the power status of first part of HP memory
+	 * to check whether it has been powered up. A few
+	 * cycles are needed for it to be powered up
+	 */
+	status = io_reg_read(HSPGISTS0);
+	while (status != ~operation_data->ebb_new_mask0) {
+		idelay(delay_count);
+		status = io_reg_read(HSPGISTS0);
+	}
+	/* Query the power status of second part of HP memory */
+
+	status = io_reg_read(HSPGISTS1);
+	while (status != ~operation_data->ebb_new_mask1) {
+		idelay(delay_count);
+		status = io_reg_read(HSPGISTS1);
+	}
+	/* add some delay before touch power register */
+	idelay(delay_count);
+}
+
+/**
+ * \brief Set memory power gating hw bit mask for memory banks
+ * \param[in] start_bank_id Id of first bank to be managed (inclusive) 0 based.
+ * \param[in] ending_bank_id Id of last bank to be managed (inclusive) 0 based.
+ * \param[in] bit Boolean deciding banks desired state (1 powered 0 gated).
+ */
+static void set_banks_gating(uint32_t start_bank_id, uint32_t ending_bank_id,
+			     uint32_t bit)
```

Comment:
bit should be renamed to enabled.

Reply:
This function sets either power gating to on or off for banks. I don't see how it can be renamed to enable.